perllib/FixMyStreet/Cobrand/Southwark.pm

```diff
@@ -30,4 +32,168 @@ sub disambiguate_location {
+        buffer => 50, # metres
+        url => "https://$host/mapserver/southwark",
+        srsname => "urn:ogc:def:crs:EPSG::27700",
+        typename => "LSG",
+        property => "USRN",
+        accept_feature => sub {
+            # Roads that only have a number, not a name, mustn't be used for
+            # site codes as they're not something Southwark can deal with.
+            # For example "A201", "A3202".
+            my $feature = shift;
+            my $name = $feature->{properties}->{Street_or_numbered_street} || "";
+            return ( $name =~ /^A[\d]+/ ) ? 0 : 1;
+        }
+    };
+}
+
+sub report_new_is_in_estate {
+    my ( $self ) = @_;
+
+    return $self->estate_feature_for_point(
+        $self->{c}->stash->{latitude},
+        $self->{c}->stash->{longitude}
+    ) ? 1 : 0;
+}
+
+sub estate_feature_for_point {
+    my ( $self, $lat, $lon ) = @_;
+
+    my ($x, $y) = Utils::convert_latlon_to_en($lat, $lon, 'G');
+
+    my $host = FixMyStreet->config('STAGING_SITE') ? "tilma.staging.mysociety.org" : "tilma.mysociety.org";
+    my $cfg = {
+        url => "https://$host/mapserver/southwark",
+        srsname => "urn:ogc:def:crs:EPSG::27700",
+        typename => "Estates",
+        filter => "<Filter><Contains><PropertyName>Geometry</PropertyName><gml:Point><gml:coordinates>$x,$y</gml:coordinates></gml:Point></Contains></Filter>",
+    };
+
+    my $features = $self->_fetch_features($cfg, $x, $y);
+    return $features->[0];
+}
+
+# Southwark do not want certain TfL categories to appear, dependent on
+# whether an 'estates' or 'street' area has been selected
+sub category_groups_to_skip {
+    return {
+        estates => [
+            'Bus Stations',
+            'Bus Stops and Shelters',
+            'River Piers',
+            'Traffic Lights',
+        ],
+        street => [
+            'River Piers',
+        ],
+    };
+}
+
+sub munge_categories {
+    my ( $self, $contacts ) = @_;
+
+    if ( $self->report_new_is_in_estate ) {
+        @$contacts = grep {
+            $_->email !~ /^STCL_/;
+        } @$contacts;
+
+        @$contacts = _filter_categories_by_group( $contacts, 'estates' );
+    } else {
+        @$contacts = grep {
+            $_->email !~ /^HOU_/;
+        } @$contacts;
+
+        @$contacts = _filter_categories_by_group( $contacts, 'street' );
+    }
+}
+
+sub _filter_categories_by_group {
+    # $area_type is either 'estates' or 'street'
+    my ( $contacts, $area_type ) = @_;
+
+    my %contacts_hash = map { $_->category => $_ } @$contacts;
+
+    for my $contact ( values %contacts_hash ) {
+        for my $group ( @{ $contact->groups } ) {
+            if ( any { $_ eq $group }
+                @{ category_groups_to_skip()->{$area_type} } )
+            {
+                delete $contacts_hash{ $contact->category };
+                last;
+            }
+        }
+    }
+
+    return values %contacts_hash;
+}
+
+sub allow_anonymous_reports { 'button' }
+
+sub anonymous_account {
```

Comment:
Since you wrote this probably, the default code now handles this, so this function shouldn't be needed any more.